### PR TITLE
super-slicer: fix build with GCC 14

### PR DESCRIPTION
**Pull request by FliegendeWurst**
Continued from #361477

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bendlas** [comment]
I'm getting a build error, when applying this on latest unstable:

```console
± nix-build . -A super-slicer-latest
this derivation will be built:
  /nix/store/47j1v340k32jn3v46bjkzqi9k42xdzlr-super-slicer-2.4.58.5.drv
building '/nix/store/47j1v340k32jn3v46bjkzqi9k42xdzlr-super-slicer-2.4.58.5.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/vfqwfnc0m23mlndi80amcfagyk3dddrn-source
source root is source
Running phase: patchPhase
substituteStream() in derivation super-slicer-2.4.58.5: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'src/PrusaSlicer.cpp')
substituteStream() in derivation super-slicer-2.4.58.5: WARNING: pattern list\(APPEND\ wxWidgets_LIBRARIES\ libexpat\) doesn't match anything in file 'src/CMakeLists.txt'
substituteStream() in derivation super-slicer-2.4.58.5: WARNING: pattern libexpat doesn't match anything in file 'src/libslic3r/CMakeLists.txt'
applying patch /nix/store/a1vdclpnypig8j0n419590qjk9vgqdki-superslicer-2.4.58.3-boost-1.79-port-v2.patch
patching file src/hints/HintsToPot.cpp
patching file src/libslic3r/AppConfig.cpp
patching file src/libslic3r/AppConfig.hpp
patching file src/libslic3r/LocalesUtils.cpp
patching file src/libslic3r/Preset.cpp
patching file src/slic3r/GUI/CreateMMUTiledCanvas.cpp
patching file src/slic3r/GUI/DesktopIntegrationDialog.cpp
patching file src/slic3r/GUI/GUI_App.cpp
patching file src/slic3r/GUI/HintNotification.cpp
patching file src/slic3r/GUI/Preferences.cpp
patching file src/slic3r/GUI/ScriptExecutor.cpp
applying patch /nix/store/xgd5yzxl0ln024327ipsz2kdjmdslcwf-meshboolean-const.patch
patching file src/libslic3r/MeshBoolean.cpp
Hunk #1 FAILED at 147.
1 out of 1 hunk FAILED -- saving rejects to file src/libslic3r/MeshBoolean.cpp.rej
error: builder for '/nix/store/47j1v340k32jn3v46bjkzqi9k42xdzlr-super-slicer-2.4.58.5.drv' failed with exit code 1;
       last 23 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/vfqwfnc0m23mlndi80amcfagyk3dddrn-source
       > source root is source
       > Running phase: patchPhase
       > substituteStream() in derivation super-slicer-2.4.58.5: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'src/PrusaSlicer.cpp')
       > substituteStream() in derivation super-slicer-2.4.58.5: WARNING: pattern list\(APPEND\ wxWidgets_LIBRARIES\ libexpat\) doesn't match anything in file 'src/CMakeLists.txt'
       > substituteStream() in derivation super-slicer-2.4.58.5: WARNING: pattern libexpat doesn't match anything in file 'src/libslic3r/CMakeLists.txt'
       > applying patch /nix/store/a1vdclpnypig8j0n419590qjk9vgqdki-superslicer-2.4.58.3-boost-1.79-port-v2.patch
       > patching file src/hints/HintsToPot.cpp
       > patching file src/libslic3r/AppConfig.cpp
       > patching file src/libslic3r/AppConfig.hpp
       > patching file src/libslic3r/LocalesUtils.cpp
       > patching file src/libslic3r/Preset.cpp
       > patching file src/slic3r/GUI/CreateMMUTiledCanvas.cpp
       > patching file src/slic3r/GUI/DesktopIntegrationDialog.cpp
       > patching file src/slic3r/GUI/GUI_App.cpp
       > patching file src/slic3r/GUI/HintNotification.cpp
       > patching file src/slic3r/GUI/Preferences.cpp
       > patching file src/slic3r/GUI/ScriptExecutor.cpp
       > applying patch /nix/store/xgd5yzxl0ln024327ipsz2kdjmdslcwf-meshboolean-const.patch
       > patching file src/libslic3r/MeshBoolean.cpp
       > Hunk #1 FAILED at 147.
       > 1 out of 1 hunk FAILED -- saving rejects to file src/libslic3r/MeshBoolean.cpp.rej
       For full logs, run 'nix log /nix/store/47j1v340k32jn3v46bjkzqi9k42xdzlr-super-slicer-2.4.58.5.drv'.
```

**Comment by FliegendeWurst** [comment]
I moved the custom patches to postPatch.

File: pkgs/applications/misc/prusa-slicer/super-slicer.nix
Comment (Aleksanaa) [inline pkgs/applications/misc/prusa-slicer/super-slicer.nix:57]: This is probably wrong?

**Comment by srounce** [comment]
I've found that the `postInstall` is also incorrect for super-slicer packages.

It should be, otherwise the OCCTWrapper.so that is output to the `bin` directory will fail to load. This is done in the original prusa-slicer derivation but `postInstall` is set to `null` in super-slicer. 
```nix
    postInstall = ''
      mkdir -p "$out/lib"
      mv -v $out/bin/*.so $out/lib/
    '';
```

I'd submit a PR for this but given the build is currently broken I'll hold off until this PR is merged.

**Review by srounce — APPROVED** [review]
All three attributes (`super-slicer`, `super-slicer-beta`, `super-slicer-latest`) are building and functional again.

**Comment by bendlas** [comment]
Works. Thanks for the fix!

**Pull request merged**
